vms/evm/predicate/predicate_results_test.go

```diff
@@ -0,0 +1,128 @@
+		},
+		"single tx multiple results": {
+			results: map[common.Hash]TxResults{
+				{1}: map[common.Address][]byte{
+					{2}: {1, 2, 3},
+					{3}: {1, 2, 3},
+				},
+			},
+			expectedHex: "000000000001010000000000000000000000000000000000000000000000000000000000000000000002020000000000000000000000000000000000000000000003010203030000000000000000000000000000000000000000000003010203",
+		},
+		"multiple txs no result": {
+			results: map[common.Hash]TxResults{
+				{1}: map[common.Address][]byte{},
+				{2}: map[common.Address][]byte{},
+			},
+			expectedHex: "000000000002010000000000000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000000",
+		},
+		"multiple txs single result": {
+			results: map[common.Hash]TxResults{
+				{1}: map[common.Address][]byte{
+					{2}: {1, 2, 3},
+				},
+				{2}: map[common.Address][]byte{
+					{3}: {3, 2, 1},
+				},
+			},
+			expectedHex: "000000000002010000000000000000000000000000000000000000000000000000000000000000000001020000000000000000000000000000000000000000000003010203020000000000000000000000000000000000000000000000000000000000000000000001030000000000000000000000000000000000000000000003030201",
+		},
+		"multiple txs multiple results": {
+			results: map[common.Hash]TxResults{
+				{1}: map[common.Address][]byte{
+					{2}: {1, 2, 3},
+					{3}: {3, 2, 1},
+				},
+				{2}: map[common.Address][]byte{
+					{2}: {1, 2, 3},
+					{3}: {3, 2, 1},
+				},
+			},
+			expectedHex: "000000000002010000000000000000000000000000000000000000000000000000000000000000000002020000000000000000000000000000000000000000000003010203030000000000000000000000000000000000000000000003030201020000000000000000000000000000000000000000000000000000000000000000000002020000000000000000000000000000000000000000000003010203030000000000000000000000000000000000000000000003030201",
+		},
+		"multiple txs mixed results": {
+			results: map[common.Hash]TxResults{
+				{1}: map[common.Address][]byte{
+					{2}: {1, 2, 3},
+				},
+				{2}: map[common.Address][]byte{
+					{2}: {1, 2, 3},
+					{3}: {3, 2, 1},
+				},
+				{3}: map[common.Address][]byte{},
+			},
+			expectedHex: "000000000003010000000000000000000000000000000000000000000000000000000000000000000001020000000000000000000000000000000000000000000003010203020000000000000000000000000000000000000000000000000000000000000000000002020000000000000000000000000000000000000000000003010203030000000000000000000000000000000000000000000003030201030000000000000000000000000000000000000000000000000000000000000000000000",
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			require := require.New(t)
+			predicateResults := NewResultsFromMap(test.results)
+			b, err := predicateResults.Bytes()
+			require.NoError(err)
+
+			parsedPredicateResults, err := ParseResults(b)
+			require.NoError(err)
+			require.Equal(predicateResults, parsedPredicateResults)
+			require.Equal(test.expectedHex, common.Bytes2Hex(b))
```

Comment:
I think this should be moved to after `predicateResults.Bytes()`, rather than where is currently is (after `ParseResults(b)`)